tinymce-per-block/src/blocks/image-block/index.js

```diff
@@ -72,5 +72,13 @@ registerBlock( 'image', {
 			attrs: { /* caption: block.caption ,*/ align: block.align },
 			rawContent
 		};
+	},
+	create: () => {
+		return {
```

Comment:
Yeah, transforming is tricky 😄  I'd put some thought into it, but didn't make a whole lot of headway. One problem is trying to find a common "base content" type (`rawContent` in your snippet). Is it a string? How do blocks declare whether they can support transforming?

I'd also wondered since we'll need to preserve backwards compatibility, and perhaps to avoid treating paragraphs as formal blocks, a block could register patterns that it can identify and be instantiated from.

Some rough pseudo-code:

```js
match: [
	{
		pattern: wp.shortcode.regexp( 'gallery' ),
		// RegExp -> Element
		initialize: function( match ) {}
	},
	{
		selector: 'p',
		// NodeList -> Element
		initialize: function(  ) {}
	},
	{
		block: 'foo',
		// WPBlock -> Element
		initialize: function() {}
	}
}